src/point_one/fusion_engine/parsers/fusion_engine_framer.cc

```diff
@@ -59,6 +59,16 @@ void FusionEngineFramer::SetBuffer(void* buffer, size_t capacity_bytes) {
                << " B]";
     return;
   }
+  // Restrict the buffer capacity to 2^31 bytes. We don't expect to ever have a
+  // single message anywhere near that large, and don't expect users to ever
+  // pass in a buffer that size. Using uint32_t instead of size_t internally
+  // makes it easier to guarantee behavior between 64b and 32b architectures. We
+  // do 2^31, not 2^32, so we can use int32_t for return values internally.
+  else if (sizeof(size_t) == 8 && capacity_bytes > 0x7FFFFFFF) {
```

Comment:
My strong preference is to limit this to the smaller size regardless of machine architecture.

Reply:
Oh good point. I originally limited to 2^32 so this if was irrelevant on 32b, hence the size check. But then I had to limit to 31 so we could use negative returns internally and forgot to remove the size limit. Will fix.